src/wallet/wallet.cpp

```diff
@@ -5186,14 +5186,21 @@ void CWallet::AvailableCoins(vector<COutput>& vCoins,
             if (nDepth < nMinDepth)
                 continue;
 
+            // if coincontrol allows watch-only addresses, add ISMINE_WATCH_SOLVABLE to the filter
+            const auto watchConditionedFilter =
+                (coinControl != nullptr && coinControl->fAllowWatchOnly) ?
+                    (filter | ISMINE_WATCH_SOLVABLE) :
+                    filter;
+
             for (unsigned int i = 0; i < pcoin.vout.size(); i++) {
                 const auto& output = pcoin.vout[i];
                 isminetype mine = IsMine(output);
 
-                bool isSpendable = ((mine & ISMINE_SPENDABLE) != ISMINE_NO) ||
-                                    (coinControl && coinControl->fAllowWatchOnly && (mine & ISMINE_WATCH_SOLVABLE) != ISMINE_NO);
+                CTxDestination addr;
+                ExtractDestination(output.scriptPubKey, addr);
 
```

Comment:
I think these lines failed to be removed during some earlier rebase, and are redundant now that this PR is based on `master` which includes #6467.
```suggestion
```